pkg/runtime/runtime.go

```diff
@@ -342,7 +343,7 @@ func newDaprRuntime(ctx context.Context,
 			for _, comp := range comps {
 				go func(comp compapi.Component) {
 					log.Infof("Shutting down component %s", comp.LogName())
-					errCh <- rt.processor.Close(comp)
+					errCh <- rt.processor.Close(ctx, comp)
```

Comment:
@JoshVanL here, easier to use `ctx` passed in to `newDaprRuntime` but perhaps it should be receiving it from the `AddCloser` func instead

Reply:
We can't use this context because it will have been cancelled when shutting down, meaning it won't actually send the close  report. I think instead we actually want to use a `context.Background()` with a `5s` timeout. Sorry!

Reply:
And to be clear, Close() should revert to not accept a context like before, and the Background be made just before the report call.